### PR TITLE
Actually rename and delete parameters in function editor

### PR DESCRIPTION
**Pull request by balderdash**
Selecting "Rename parameter..." or "Delete parameter..." after clicking on a parameter in the modal function editor actually does what the dialog boxes say they will do. Renaming changes every reference to the parameter, and deleting removes all of them.

### CODE REVIEW

File: blockly-core/core/ui/function_editor.js [hunk 1]
```diff
@@ -283,6 +286,9 @@ Blockly.FunctionEditor.prototype.removeParameter = function(nameToRemove) {
   });
   keysToDelete.forEach(function(key) { this.orderedParamIDsToBlocks_.remove(key); }, this);
   this.refreshParamsEverywhere();
+  this.forEachParameterGetBlock(nameToRemove, function(block) {
```
Comment (Hamms) [inline blockly-core/core/ui/function_editor.js:288]: should this go before `this.refreshParamsEverywhere()`?

File: blockly-core/core/ui/function_editor.js [hunk 2]
```diff
@@ -328,6 +334,15 @@ Blockly.FunctionEditor.prototype.paramsAsParallelArrays_ = function() {
   return {paramNames: paramNames, paramIDs: paramIDs, paramTypes: paramTypes};
 };
 
+Blockly.FunctionEditor.prototype.forEachParameterGetBlock = function(paramName, callback) {
```
Comment (Hamms) [inline blockly-core/core/ui/function_editor.js:337]: Nice helper method!

File: blockly-core/tests/modal_editors_test.js [hunk 3]
```diff
@@ -92,6 +136,17 @@ function openFunctionEditor(opt_name) {
   Blockly.functionEditor.autoOpenFunction(opt_name || 'test-function');
 }
 
+function getParametersUsedInFunctionEditor() {
+  var paramsUsed = []
+  Blockly.functionEditor.functionDefinitionBlock.getDescendants().forEach(
```
Comment (Hamms) [inline blockly-core/tests/modal_editors_test.js]: this could also accomplished with a `filter` and a `map`, but this certainly works

**Comment by Hamms** [comment]
Great tests! Looks good to me after notes

File: blockly-core/core/ui/function_editor.js (hunk 1 above)
Comment (balderdash) [inline blockly-core/core/ui/function_editor.js:288]: Yep, done.

File: blockly-core/tests/modal_editors_test.js (hunk 3 above)
Comment (balderdash) [inline blockly-core/tests/modal_editors_test.js]: That's definitely nicer.

**Pull request merged**